tests/components/mealie/test_services.py

```diff
@@ -297,77 +231,128 @@ async def test_service_set_random_mealplan(
     hass: HomeAssistant,
     mock_mealie_client: AsyncMock,
     mock_config_entry: MockConfigEntry,
+    service: str,
+    payload: dict[str, str],
+    function: str,
+    exception: Exception,
+    raised_exception: type[Exception],
+    message: str,
 ) -> None:
-    """Test a connection error in the get_mealplans service."""
+    """Test a connection error in the services."""
 
     await setup_integration(hass, mock_config_entry)
 
-    mock_mealie_client.get_mealplans.side_effect = MealieConnectionError
+    getattr(mock_mealie_client, function).side_effect = exception
 
-    with pytest.raises(HomeAssistantError):
+    with pytest.raises(raised_exception, match=message):
         await hass.services.async_call(
             DOMAIN,
-            SERVICE_GET_MEALPLAN,
-            {ATTR_CONFIG_ENTRY_ID: mock_config_entry.entry_id},
+            service,
+            {ATTR_CONFIG_ENTRY_ID: mock_config_entry.entry_id} | payload,
             blocking=True,
             return_response=True,
         )
 
 
-async def test_service_mealplan_without_entry(
+@pytest.mark.parametrize(
+    ("service", "payload"),
+    [
+        (SERVICE_GET_MEALPLAN, {}),
+        (SERVICE_GET_RECIPE, {ATTR_RECIPE_ID: "recipe_id"}),
+        (SERVICE_IMPORT_RECIPE, {ATTR_URL: "http://example.com"}),
+        (
+            SERVICE_SET_RANDOM_MEALPLAN,
+            {ATTR_DATE: "2023-10-21", ATTR_ENTRY_TYPE: "lunch"},
+        ),
+    ],
+)
+async def test_service_entry_availability(
     hass: HomeAssistant,
+    mock_mealie_client: AsyncMock,
     mock_config_entry: MockConfigEntry,
+    service: str,
+    payload: dict[str, str],
 ) -> None:
-    """Test the get_mealplan service without entry."""
+    """Test the services without valid entry."""
     mock_config_entry.add_to_hass(hass)
     mock_config_entry2 = MockConfigEntry(domain=DOMAIN)
     mock_config_entry2.add_to_hass(hass)
     await hass.config_entries.async_setup(mock_config_entry.entry_id)
     await hass.async_block_till_done()
 
-    with pytest.raises(ServiceValidationError):
+    with pytest.raises(ServiceValidationError, match="Mock Title is not loaded"):
         await hass.services.async_call(
             DOMAIN,
-            SERVICE_GET_MEALPLAN,
-            {ATTR_CONFIG_ENTRY_ID: mock_config_entry2.entry_id},
+            service,
+            {ATTR_CONFIG_ENTRY_ID: mock_config_entry2.entry_id} | payload,
             blocking=True,
             return_response=True,
         )
 
-    with pytest.raises(ServiceValidationError):
+    with pytest.raises(
+        ServiceValidationError, match='Integration "mealie" not found in registry'
+    ):
         await hass.services.async_call(
             DOMAIN,
-            SERVICE_GET_MEALPLAN,
-            {ATTR_CONFIG_ENTRY_ID: "bad-config_id"},
+            service,
+            {ATTR_CONFIG_ENTRY_ID: "bad-config_id"} | payload,
             blocking=True,
             return_response=True,
         )
```

Comment:
Shouldn't these tests assert that the api call happen?

Reply:
The api call doesn't happen as this is testing how the integration handles unavailable config entry or unloaded config entry